### PR TITLE
Chore: rewrite parseListConfig for a small perf gain.

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**

**Is there anything you'd like reviewers to focus on?**

perf tests result (~1.7x):

image

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -145,17 +145,15 @@ function parseJsonConfig(string, location) {
  */
 function parseListConfig(string) {
     const items = {};
+    const reg = /[\w-/]+/g;
+    const rs = string.match(reg);
```
Comment (not-an-aardvark) [inline lib/linter.js]: Can you give this variable a more descriptive name? The name `rs` doesn't communicate what the variable is doing very well.

Maybe something like `elementMatches` would be better.

File: lib/linter.js [hunk 2]
```diff
@@ -145,17 +145,15 @@ function parseJsonConfig(string, location) {
  */
 function parseListConfig(string) {
     const items = {};
+    const reg = /[\w-/]+/g;
```
Comment (platinumazure) [inline lib/linter.js]: Should this hyphen be escaped?

Comment (Aladdin-ADD) [reply lib/linter.js]: no need in `[]`

**Comment by platinumazure** [comment]
It has special meaning in [a-z] but maybe not when it is preceded by \w. I
would feel better seeing it after [ or before ]. If there are negative
tests for this logic then I'm okay with no change. Thanks!

File: lib/linter.js (hunk 2 above)
Comment (Aladdin-ADD) [reply lib/linter.js]: oh, before `/` , it's no need. I was confused.

**Pull request merged**